syntax = "proto3";

import "google/protobuf/timestamp.proto";

message ComplexHeader {
    repeated string values = 1;
    google.protobuf.Timestamp timestamp = 2;
    map<string, string> details = 3;
}

message ComplexData {
    map<string, ComplexHeader> headers = 1;
    double start_time = 2;
}

service HeaderService {
    rpc SendComplexData(ComplexData) returns (Empty);
}

message Empty {}